src/server/user-controller.js

```diff
@@ -239,11 +239,12 @@ class UserController {
         await this.requestLimitRecordService.resetCount({ req });
       }
 
-      return req.login(user, loginError => {
+      return req.login(user, async loginError => {
         if (loginError) {
           return next(loginError);
         }
 
+        await this.userService.recordUserLogIn(user._id);
```

Comment:
This only updates the user in the database while the `user` object used here for the request still is the "old" version. What do you think of returning the updated user from the services end use that one instead of `user` from here on?

Reply:
What do you mean by "directly on the user"?
Technically we don't use the `lastLoggedIn` date in this context at all, but makes sense to return the updated value here.

Reply:
Sorry, with the edit it got confusing. I try to phrase it clearer: By calling `req.login` we're basically setting `req.user`. But we call it with the not-updated user (it has the non-current `lastLoggedIn` value). I thought about it on the way home and realized that calling `req.login` basically never fails (the error handling code is just there out of formality), so I'd suggest to call `req.login` with an already updated `user` object (that maybe for example could be returned from `recordUserLogIn`. So the `req.user` reflects the current database state. (I know we do not use the property anywhere, nevertheless I think it should be the most up to date object)

Reply:
Fully understood what you meant now, thanks for clarifying. Done it like that.